sql-migrate/config.go

```diff
@@ -60,9 +74,19 @@ func ReadEnvFromFile() (*Environment, error) {
 }
 
 func GetEnvironment() (*Environment, error) {
-	env, err := ReadEnvFromFile()
-	if err != nil {
-		return nil, err
+	var env *Environment
+	if ConfigPresent() {
+		var err error
+		env, err = ReadEnvFromFile()
+		if err != nil {
+			return nil, err
+		}
+	} else {
```

Comment:
if im keeping the order of operations right, we're favoring values coming out of the config file.  should we be though?  wouldn't we want to allow for the flexibility of command-line overrides?

Reply:
To keep this backwards compatible with the upstream project I have the cascading in this order. If we are determined enough to deviate from upstream by dropping the config functionality working by default we can change this; that was the reasoning though. Thoughts with the upstream concerns in mind?

Reply:
ah, that makes sense.  I think it may be worth pushing this question back upstream to the maintainers and see how they feel about it.  im in favor of minimizing behavior/functional drift from upstream, so this as is is good to go

Reply:
as part of trying to get this upstream I will see how they feel about the env reading in general and if it would be sane enough to try to read from env first.